Sources/MapboxCoreNavigation/RouteController.swift

```diff
@@ -570,6 +570,11 @@ open class RouteController: NSObject {
         BillingHandler.shared.stopBillingSession(with: sessionUUID)
         unsubscribeNotifications()
         routeTask?.cancel()
+        Navigator.shared.navigator.setRoutesFor(nil) { _ in
```

Comment:
For drop-in UI, this `deinit` happens when the view controller is deallocated, which might happen after passive navigation is started. Is there a way to call this earlier?

Reply:
I don't know how to call it earlier unless we create an explicit method. 

IMHO, We shall improve FD/AG transition similar to the Android way in v3. These modes should be tightly coupled as our underlying library is.

Reply:
@1ec5 please review this code, maybe you have some suggestions.

Reply:
Would it be possible to do some of this teardown when canceling or arriving? If there needs to be an explicit method, some kind of counterpart to `updateRoute(with:routeOptions:completion:)`, that wouldn’t be so bad as long as NavigationViewController would normally take care of calling it for the developer. Changing the behavior in this way wouldn’t necessarily be a breaking change: if an application uses RouteController directly but doesn’t call the explicit teardown method, it would just mean that #3298 would go unfixed for the application, right?

By the way, `MapboxCoreNavigation.Navigator` should encapsulate `MapboxNavigationNative.Navigator` well enough that RouteController shouldn’t have to root around `MapboxCoreNavigation.Navigator.navigator`.

> IMHO, We shall improve FD/AG transition similar to the Android way in v3. These modes should be tightly coupled as our underlying library is.

Broadly speaking, I would normally say the reverse: maybe MapboxNavigationNative needs to be less tightly coupled in the long run rather than forcing MapboxCoreNavigation to be more tightly coupled.

A developer might reasonably expect to be able to initialize a view controller and hang onto it for a while before presenting it. The more we tie underlying behavior to the lifecycle of the view controller object, the more we open ourselves up to side effects when the view controller leaks, either due to a bug in the SDK or unanticipated application behavior.

Reply:
> Broadly speaking, I would normally say the reverse: maybe MapboxNavigationNative needs to be less tightly coupled in the long run rather than forcing MapboxCoreNavigation to be more tightly coupled.

> A developer might reasonably expect to be able to initialize a view controller and hang onto it for a while before presenting it. The more we tie underlying behavior to the lifecycle of the view controller object, the more we open ourselves up to side effects when the view controller leaks, either due to a bug in the SDK or unanticipated application behavior.

For you information: @mapbox/navnative

Reply:
> Would it be possible to do some of this teardown when canceling or arriving? If there needs to be an explicit method, some kind of counterpart to updateRoute(with:routeOptions:completion:), that wouldn’t be so bad as long as NavigationViewController would normally take care of calling it for the developer. 

I'll try. It should be possible, yes. 

> Changing the behavior in this way wouldn’t necessarily be a breaking change: if an application uses RouteController directly but doesn’t call the explicit teardown method, it would just mean that #3298 would go unfixed for the application, right?

Yes, I think so. 

> By the way, MapboxCoreNavigation.Navigator should encapsulate MapboxNavigationNative.Navigator well enough that RouteController shouldn’t have to root around MapboxCoreNavigation.Navigator.navigator.

Agree.

Reply:
Updated.